common/src/main/java/com/duckduckgo/app/global/FragmentViewModelFactory.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.global
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.duckduckgo.app.global.plugins.view_model.FragmentViewModelFactoryPluginPoint
+import com.duckduckgo.di.scopes.FragmentScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@SingleInstanceIn(FragmentScope::class)
+@ContributesBinding(FragmentScope::class)
```

Comment:
Maybe you can remove the `ContributesBinding` to ensure it goes into the DI graph as `FragmentViewModelFactory`

Reply:
@aitorvs Hmm the thing is I am injecting this as `ViewModelProvider.NewInstanceFactory` in `DuckDuckGoActivity` https://github.com/duckduckgo/Android/blob/16133270a7caf2be780b4e6df24f9799672c5515/common-ui/src/main/java/com/duckduckgo/app/global/DuckDuckGoActivity.kt#L36 I think I need this to tell Dagger to bind `FragmentViewModelFactory` as `ViewModelProvider.NewInstanceFactory`.